java/internal/src/cuvs_java.c

```diff
@@ -91,3 +93,33 @@ void search_cagra_index(cuvsCagraIndex_t index, float *queries, int topk, long n
   cudaMemcpy(neighbors_h, neighbors, sizeof(uint32_t) * n_queries * topk, cudaMemcpyDefault);
   cudaMemcpy(distances_h, distances, sizeof(float) * n_queries * topk, cudaMemcpyDefault);
 }
+
+typedef struct {
+    char name[64];
+    size_t totalMemory;
+    size_t freeMemory;
+} GpuDetail;
+
+int get_gpu_details(GpuDetail *details, int maxGpus) {
+    int deviceCount = 0;
+    cudaError_t err = cudaGetDeviceCount(&deviceCount);
+
+    if (err != cudaSuccess || deviceCount == 0) {
```

Comment:
Multiple problems here: [1] how can we return -1 when `deviceCount` is 0? If I remember -1 signifies an error. [2] Also, even if it made sense logically, do you think the control will ever reach the else-if block if the `deviceCount` is 0?

Reply:
True, changed this part of the code, had missed this.